### DISCUSSION TITLE
Feature request: auto layout

### DISCUSSION THREAD

**Discussion by littlesmilelove** [bcakmakoglu/vue-flow#81] [Ideas]
In the previous version, you mentioned using `dagre` to do auto layout, but in the recent version, I can't find this piece of content. And in my practice, `dagre` doesn't seem to work with custom nodes.
So, do you have plans to support auto layout?

**Comment by bcakmakoglu** [comment]
Hey,

The dagre example is still available [right here](https://github.com/bcakmakoglu/vue-flow/tree/master/examples/vite/src/Layouting), it's just not on the documentation site anymore.

I proposed the idea of a layout feature in #61, so it is something that I plan on implementing but I can't say when - probably with the next major (0.5.0) release.

**Reply by tuandunguit** [reply]
@bcakmakoglu the url to dagre example is deleted, is it moved to the new place?

**Reply by bcakmakoglu** [reply]
@tuandunguit Yeah let me update the link 😄

**Reply by bcakmakoglu** [reply]
Updated - now it points to the correct example path

**Reply by tuandunguit** [reply]
great, thank you

**Comment by brandonreid** [comment]
I was imagining a way where each stage of nodes could be rendered with flexbox styling, then all the stages would be in a larger flexbox container. You could set the gaps between items easily and the content would be free to have it's own sizing. [Here's a codepan](https://codepen.io/brandonreid/pen/RwJwgxd).

If the edges are applied after, they can measure from each element.

**Reply by bcakmakoglu** [reply]
I see what you mean, the idea isn't bad but I'm not sure if at this point it's the right step to take with the core library as it entails a severe change in how everything works and how users expect it to work right now.
Instead it might be better to offer alternative "renderers" that could handle this sort of thing for you depending on how you want the graph to be laid out.

I'll have to think about this, though I thank you very much for the codepen :) It's a good example.
